design/kiota-e2e.md

```diff
@@ -0,0 +1,147 @@
+
+## Constraints
+
+Before we jump into the end-to-end walk-through, it's important to set some constraints.
+
+| Type      | Description                                                                                                                                   |
+| --------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| Platforms | Node : Current and Previous LTS (14 / 16)<br /> Web : Edge, Chrome, Firefox and Safari (Latest released version + immediate previous version) |
+| Modules   | Node : CommonJS<br /> Web : ES Module (ES6)                                                                                                   |
+| Types     | Typings should be available for both the core and the service libraries for Graph models                                                      |
+
+## NodeJS e2e using the Service library
+
+```bash
+npm install @microsoft/msgraph-sdk-typescript --save
+npm install @microsoft/kiota-authentication-azure --save
+```
+
+```typescript
+// App.ts
+
+import { Client, User, Message, BodyType } from "@microsoft/msgraph-sdk-javascript";
+import { AzureIdentityAuthenticationProvider } from "@microsoft/kiota-authentication-azure";
+import { DeviceCodeCredential } from "@azure/identity";
+
+const deviceCodeCredentials = new DeviceCodeCredential({
+	tenantId: "b61f9af1-d6cf-4cc0-a6f6-befb38bc00ed",
+	clientId: "bde251a6-0ef9-42a8-a40b-9ad9bb594b2c",
+});
+
+const scopes = ["User.Read", "Mail.Send"];
+
+const graphClient = Client.init({
+	accessTokenProvider: new AzureIdentityAccessTokenProvider(deviceCodeCredentials, scopes),
+});
+
+const me = await getMe();
+const meRaw = await getMeRaw();
+await sendMail();
+await sendMailRaw();
+
+async function getMe(): Promise<User | undefined> {
+	return await graphClient.me.get();
+}
+
+async function getMe(): Promise<User | undefined> {
+	return await graphClient.api("/me").get();
+}
+
+async function sendMail(): Promise<void> {
+	const message: Message = {
+		subject: "Hello Graph TypeScript SDK!",
+		body: {
+			contentType: BodyType.Html,
+			content: "<bold>Hello Graph TypeScript SDK!</bold>",
+		},
+		toRecipients: [
+			{
+				emailAddress: {
+					address: "admin@m365x263716.onmicrosoft.com",
+				},
+			},
+		],
+	};
+
+	return await client.me.sendMail.post(message);
+}
+
+async function sendMailRaw(): Promise<void> {
+	const message: Message = {
+		subject: "Hello Graph TypeScript SDK!",
+		body: {
+			contentType: BodyType.Html,
+			content: "<bold>Hello Graph TypeScript SDK!</bold>",
+		},
+		toRecipients: [
+			{
+				emailAddress: {
+					address: "admin@m365x263716.onmicrosoft.com",
+				},
+			},
+		],
+	};
+
+	return await client.api("/me/sendMail").post({
+		message: message,
+	});
+}
+```
+
+## NodeJS e2e using the Core library
+
+```bash
```

Comment:
maybe I'm missing something but it feels like what we're trying to show here is already covered by the previous example?

Reply:
Not exactly. Here we're highlighting that the types should be separate from the service library to allow using the same types with the core (without the service library). Very similar, but I feel is required to understand the problem.

Reply:
maybe add some comments in the code to call out the differences, or use a diff code block